Objects/listobject.c

```diff
@@ -559,9 +559,9 @@ list_repeat(PyListObject *a, Py_ssize_t n)
         return NULL;
     PyObject **dest = np->ob_item;
     PyObject **dest_end = dest + size;
+    PyObject *elem;
     if (Py_SIZE(a) == 1) {
-        PyObject *elem = a->ob_item[0];
-        Py_SET_REFCNT(elem, Py_REFCNT(elem) + n);
+        Py_SET_REFCNT((elem = a->ob_item[0]), Py_REFCNT(elem) + n);
```

Comment:
C doesn't have a specified order in which function arguments are evaluated (legal C compilers can evaluate left to right, right to left, or some other random order; it can even vary by optimization level on the same compiler). You can't guarantee `elem` will be assigned to before `Py_REFCNT(elem)` tries to read from it. Thankfully we're moving away from macros, but this could be even worse if it were ever reverted to a macro. Don't try to combine the two lines into one.

Reply:
Ok, got it.